src/components/Comments/ViewComments/ViewComments.js

```diff
@@ -0,0 +1,124 @@
+import React, { useState, useEffect } from 'react';
+import cookie from 'react-cookies';
+import {
+  Typography, Avatar, List, ListItem, ListItemAvatar, ListItemText,
+} from '@material-ui/core';
+import useStyles from './style';
+import { ErrorMessage } from '../../ErrorMessage';
+import { serviceRequest } from '../../../services/serviceRequest';
+import { useStateValue } from '../../StateProvider';
+import { Pagination } from '../../Pagination';
+// import config from '../../../config';
+
+// const path = '/api/v1/post/viewcomments';
+// const domain = config.apiDomain;
+
+const preparePayload = (method, headers, params) => {
+//   const url = `https://${domain}${path}`;
+  const url = 'http://localhost:8081/api/v1/post/viewcomments';
+  return {
+    method,
+    url,
+    headers,
+    params,
+  };
+};
+
+export const ViewComments = () => {
+  const classes = useStyles();
+  const [{ comments, page }, dispatch] = useStateValue();
+  const [errorInfo, setErrorInfo] = useState({ isError: false, errorMsg: null });
+
+  const urls = window.location.href;
+  const postId = urls.slice(urls.lastIndexOf('/') + 1, urls.length);
```

Comment:
This may be a better way to refactor the logic https://davidwalsh.name/query-string-javascript

Reply:
I will check it out!

Reply:
in order to use the window.location.search, it required query string as url (with `?` in the url), right now we have `/` in the url, not sure if its a good idea to change it to `?`